xml/html_title.xml

```diff
@@ -4196,4 +4196,59 @@
     <param pos="0" name="hw.device" value="Broadband Router"/>
   </fingerprint>
 
+  <!-- FatPipe Networks fingerprints -->
+
+  <fingerprint pattern="^FatPipe MPVPN(?:.nbsp;\| Log in| Remote Configuration)$">
```

Comment:
In terms of future-proofing a regex like this, i.e. if they update their page to have a different suffix - such as `FatPipe MPVPN - VPN Security | Login` or similar; would a partial match suffice?

```suggestion
  <fingerprint pattern="^FatPipe MPVPN.*$">
```

Or is that considered an anti-pattern?

Reply:
I'm a fan of future flexibility, but the mindset handed off when I joined the project is that the fingerprint should be as specific as possible and only add flexibility where you know it needs to exist, for example, if there were a hostname in the title, etc.

Perform a quick google `intitle:"FatPipe MPVPN"` and you'll see examples that would result in a false positive match with the `.*` change.
```
FatPipe MPVPN - Product Overview
FatPipe MPVPN 2U 10Gbps Application Delivery Controller
FatPipe MPVPN Reviews & Product Details
```